server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,251 @@
+/**
+* The /fee_centers HTTP API endpoint
+*
+* @module finance/feeCenters
+*
+* @description This controller is responsible for implementing all crud and others custom request
+* on the fee_center table through the `/fee_centers` endpoint.
+*
+* @requires lib/db
+**/
+
+
+var db = require('../../lib/db');
+var NotFound = require('../../lib/errors/NotFound');
+
+/**
+* Returns an array of fee centers
+*
+* @example
+* // GET /fee_centers : Get list of fee_centers
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.list(req, res, next);
+*/
+
+function list (req, res, next) {
+  'use strict';
+
+  var sql =
+    'SELECT fc.id, fc.label, fc.is_cost FROM fee_center AS fc ';
+  var filter = '', ordering = 'ORDER BY fc.label;';
+
+  if (req.query.detailed === '1') {
+    sql =
+      `SELECT fc.id, fc.label, fc.project_id, fc.is_cost, fc.note, fc.is_principal, p.name, p.abbr, p.enterprise_id, p.zs_id
+      FROM fee_center AS fc JOIN project AS p ON fc.project_id = p.id `;
+  }
+
+
+  if(req.query.available === '1') {
+    filter += ' fc.id NOT IN (SELECT s.cc_id FROM service AS s WHERE NOT ISNULL(s.cc_id)) AND fc.id NOT IN (SELECT s.pc_id FROM service AS s WHERE NOT ISNULL(s.pc_id)) ';
+  }
+
+  if(req.query.is_cost){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_cost = ${req.query.is_cost} `;
```

Comment:
I have pulled your branch typed this into my web browser.
```
http://localhost:8080/fee_centers?is_cost=1 UNION SELECT id, password as label, active AS is_cost FROM user AS fc ; --
```
And this is the result I got:
![sql injection attack](https://cloud.githubusercontent.com/assets/896472/22788931/b6495722-eee1-11e6-8233-4fdab507dfae.png)
_Fig 1: An SQL Injection Attack_

These attacks are quite common across the web.  We have a library called `db` that will parse the parameters and prevent sensitive information like passwords from leaking.

You've made this mistake before and it makes me sad to have this problem across multiple code reviews.